extension-workspace-reference.md

```diff
@@ -0,0 +1,121 @@
+
+### Initialization
+
+`ServerCapabilities` may contain a new field to indicate server-side support for this extension:
+
+```typescript
+interface ServerCapabilities {
+  /* ... all fields from the base ServerCapabilities ... */
+
+  /**
+   * The server provides workspace reference exporting support.
+   */
+  xworkspaceReferenceProvider?: boolean;
+}
+```
+
+#### Workspace Reference Request
+
+The workspace reference request is sent from the client to the server to export project-wide references to dependencies. That is, the response strictly returns references in the project to symbols defined in dependencies.
+
+_Request_
+* method: 'workspace/xreference'
+* params: `WorkspaceReferenceParams` defined as follows:
+```typescript
+/**
+ * The parameters of a Workspace Reference Request.
+ */
+interface WorkspaceReferenceParams {
+}
+```
+
+_Response_:
+* result: `ReferenceInformation[]` defined as follows:
+```typescript
+/**
+ * Represents information about a reference to programming constructs like
+ * variables, classes, interfaces etc.
+ */
+interface ReferenceInformation {
+	/**
+	 * The location in the workspace where the `symbol` has been referenced.
+	 */
+	reference: Location;
+
+	/**
+	 * Metadata information describing the symbol being referenced.
+	 */
+	symbol: ReferenceSymbolInformation;
+}
+```
+* error: code and message set in case an exception happens during the workspace reference request.
+
+Where `ReferenceSymbolInformation` is defined as follows:
+
+```typescript
+/**
+ * Represents information about a programming construct like a variable, class,
+ * interface etc that has a reference to it. Effectively, it contains data similar
+ * to SymbolInformation except all fields are optional and a metadata field is
+ * present for language-specific data.
+ *
+ * ReferenceSymbolInformation does NOT always uniquely identify a symbol. The
+ * caller should effectively consider the returned information to be
+ * information about the symbol which _generally_ (but NOT always) identifies a
+ * single symbol.
+ */
+interface ReferenceSymbolInformation {
+    /**
+     * The name of this symbol (same as `SymbolInformation.name`).
+     */
+    name?: string;
+
+    /**
+     * The kind of this symbol (same as `SymbolInformation.kind`).
+     */
+    kind?: number;
+
+    /**
+     * The file URI of this symbol (same as `SymbolInformation.location.uri`).
+     */
+    file?: string;
+
+    /**
+     * The name of the symbol containing this symbol (same as `SymbolInformation.containerName`).
+     */
+    containerName?: string;
+
+    /**
+     * The repository URI that the symbol is defined in, e.g. `github.com/golang/go`
+     * or `bitbucket.org/jespern/django-piston`.
+     */
+    repo?: string;
```

Comment:
To clarify this up a bit.. I think it could be _either_ the LS or the build server that fills this field in. In the case of Go, we don't have easy access to do that, so the build server could do it. It's optional, and we don't need the field right now -- so we don't need to. But we could do it in the build server.

In the case of a LS like PHP that has this knowledge, they can of course fill it in since it may be useful down the line :)